src/utils/hooks/useTimeline/types.ts

```diff
@@ -0,0 +1,76 @@
+import { LanguageStrings, useLanguageProvider } from '../../../contexts/Localisation'
+import { ActivityLog, EventType, useGetActivityLogQuery } from '../../generated/graphql'
+
+type GenericObject = { [key: string]: any }
+
+interface TimelineEvent {
+  id: number
+  eventType: TimelineEventType
+  timestamp: Date
+  displayString: string
+  details: { [key: string]: GenericObject }
+}
+
+interface TimelineStage {
+  number: number
+  name: string
+  colour: string
+  timestamp: Date
+  events: TimelineEvent[]
+}
+
+interface Timeline {
+  stages: TimelineStage[]
+  //   rawLog: ActivityLog[]
+}
+
+enum TimelineEventType {
```

Comment:
This is perfect! That's exactly what I was looking for when we chatted ealier. Thank you